src/lib/index/wb_cache.cpp

```diff
@@ -219,31 +242,33 @@ std::unique_ptr< CPContext > IndexWBCache::create_cp_context(cp_id_t cp_id) {
                                               m_free_blkid_list[cp_id_slot].get());
 }
 
-void IndexWBCache::do_flush_one_buf(IndexCPContext* cp_ctx, const IndexBufferPtr& buf, bool part_of_batch) {
-    LOGTRACEMOD(wbcache, "buf {}", buf->to_string());
-    buf->m_buf_state = index_buf_state_t::FLUSHING;
+void IndexWBCache::do_flush_one_buf(IndexCPContext* cp_ctx, IndexBufferPtr buf, bool part_of_batch) {
+    LOGTRACEMOD(wbcache, "cp {} buf {}", cp_ctx->id(), buf->to_string());
+    { buf->set_state(index_buf_state_t::FLUSHING); }
     m_vdev->async_write(r_cast< const char* >(buf->raw_buffer()), m_node_size, buf->m_blkid, part_of_batch)
-        .thenValue([pbuf = buf.get(), cp_ctx](auto) {
+        .thenValue([buf, cp_ctx](auto) {
```

Comment:
I am ok with this, but to note down, here is capture size goes up, which results in another malloc in the heap. I wanted to keep the capture <= 16 bytes.

Reply:
As discussed in meeting, keep the shared ptrs as they are accessed from dirty buffer and also from IndexBtreeNode